src/main/webapp/app/overview/exercise-details/course-exercise-details.component.ts

```diff
@@ -118,9 +124,53 @@ export class CourseExerciseDetailsComponent implements OnInit, OnDestroy {
         this.exercise.participationStatus = participationStatus(this.exercise);
         this.isAfterAssessmentDueDate = !this.exercise.assessmentDueDate || moment().isAfter(this.exercise.assessmentDueDate);
         this.exerciseCategories = this.exerciseService.convertExerciseCategoriesFromServer(this.exercise);
+        if (this.exercise.type.toString() === 'programming') {
+            this.getProgrammingExercise();
+        }
         this.subscribeForNewResults();
     }
 
+    /**
+     * The button should be only visible in the dev environment
+     */
+    isInProduction(): boolean {
+        return this.profileService.isInProduction();
+    }
+
+    /**
+     * checks if exercise has no local setup
+     */
+    hasNoLocalSetup(): boolean {
+        return this.programmingExerciseSimulationUtils.hasNoLocalSetup(this.programmingExercise.testRepositoryUrl);
+    }
+
+    /**
+     * asks the server for the programming exercise with the provided exercise ID
+     */
+    getProgrammingExercise() {
+        this.courseExerciseSubmissionResultSimulationService.getProgrammingExercise(this.exerciseId).subscribe((programmingExercise) => {
+            this.programmingExercise = programmingExercise;
+        });
+    }
+
+    /**
+     * triggers the simulation of a participation and submission for the currently logged in user
+     */
+    simulateSubmission() {
+        this.courseExerciseSubmissionResultSimulationService.simulateSubmission(this.exerciseId).subscribe(() => {
+            this.jhiAlertService.success('artemisApp.exercise.submissionSuccessful');
+        });
+    }
+
+    /**
+     * triggers the simulation of a result for the currently logged in user
+     */
+    simulateResult() {
+        this.courseExerciseSubmissionResultSimulationService.simulateResult(this.exerciseId).subscribe(() => {
+            this.jhiAlertService.success('artemisApp.exercise.resultCreationSuccessful');
```

Comment:
Only the sucess case is being handled. What happens if the observable emits an error?

Reply:
i added the error case